stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -398,3 +398,5 @@ Array(s::UniformScaling, dims::Dims{2}) = Matrix(s, dims)
 ## Diagonal construction from UniformScaling
 Diagonal{T}(s::UniformScaling, m::Integer) where {T} = Diagonal{T}(fill(T(s.λ), m))
 Diagonal(s::UniformScaling, m::Integer) = Diagonal{eltype(s)}(s, m)
+
+dot(x::AbstractVector, J::UniformScaling, y::AbstractVector) = J.λ * dot(x, y)
```

Comment:
This assumes `*` is commutative… Probably this should be a specialized method for `UniformScaling{<:Union{Real,Complex}}`.

Reply:
Right. I’ll add a generic `UniformScaling` version similar to the one for `Diagonal`.